### PR TITLE
Auth 4471

### PR DESCRIPTION
**Pull request by sam-ritter**

### CODE REVIEW

File: vars/buildFunctionLib.groovy [hunk 1]
```diff
@@ -3,4 +3,131 @@ import org.ithaka.auth.pipelines.buildLogStatus
...
\ No newline at end of file
+}
+
+/**
+ *
+ * @param repo    the repo name without the ".git"
+ * @param env     the enviroment to deploy to.
+ * @return
+ */
+def deployWithSagoku(String repo, String env) {
```
Comment (kavya-ithaka) [inline vars/buildFunctionLib.groovy]: can we add a branch parameter to this and use it for deployment to TEST too?

File: vars/buildFunctionLib.groovy [hunk 2]
```diff
@@ -3,4 +3,131 @@ import org.ithaka.auth.pipelines.buildLogStatus
...
+    println "Checking ${APP_NAME} deployment status..."
+    sh "sagoku.sh get -c ${EKS_CLUSTER} -d ${APP_NAME}"
+}
+
+
+def verifyPact(task, String testDir, String workSpace) {
+    println "Running ${task} Pact Verification"
+    def ret = sh(
+            returnStatus: true,
+            script: "cd ${testDir} && bundle exec rake IacService:pact:verify:${task}"
```
Comment (kavya-ithaka) [inline vars/buildFunctionLib.groovy]: this is running the pacts for IacService only - should take the service name as a param.

File: vars/buildFunctionLib.groovy (hunk 1 above)
Comment (sam-ritter) [reply vars/buildFunctionLib.groovy]: this does work for deploy to test.    this takes what ever is in the current workspace and deploys.      examples can be found in IAC:Jenkinsfile.release-* files.   we do have a ticket to combine the release files:   AUTH-4500

File: vars/buildFunctionLib.groovy (hunk 2 above)
Comment (sam-ritter) [reply vars/buildFunctionLib.groovy]: good catch!!!    fixing...

Comment (sam-ritter) [reply vars/buildFunctionLib.groovy]: changes done and ran pipeline thru deploy to test.

**Pull request merged**